File: listing-service/src/main/java/com/realestate/listing/service/AmenityService.java
// src/main/java/com/realestate/listing/service/AmenityService.java
package com.realestate.listing.service;

import com.realestate.listing.entity.Amenity;
import com.realestate.listing.entity.Property;
import com.realestate.listing.repository.AmenityRepository;
import com.realestate.listing.repository.PropertyRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
@Transactional
public class AmenityService {

    @Autowired
    private AmenityRepository amenityRepository;

    @Autowired
    private PropertyRepository propertyRepository;

    // ======================
    // === CRUD DE BASE ===
    // ======================

    /** Récupère toutes les amenities */
    public List<Amenity> getAllAmenities() {
        return amenityRepository.findAll();
    }

    /** Récupère une amenitie par ID */
    public Optional<Amenity> getAmenityById(Integer id) {
        return amenityRepository.findById(id);
    }

    /** Crée une nouvelle amenitie  */
    public Amenity createAmenute(Amenity amenity) {
        return amenityRepository.save(amenity);
    }

    /** Met à jour une amenitie existante (remplace tout) */
    public Amenity updateAmenity(Integer id, Amenity updatedAmenity) {
        Amenity existing = amenityRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Commodité non trouvée avec l'ID : " + id));

        updatedAmenity.setAmenityId(id);
        return amenityRepository.save(updatedAmenity);
    }


    /** Supprime une commodité */
    public void deleteAmenity(Integer id) {
        if (!amenityRepository.existsById(id)) {
            throw new IllegalArgumentException("Commodité non trouvée avec l'ID : " +  id);
        }
        amenityRepository.deleteById(id);
    }

    // ============================
    // === RECHERCHES SPÉCIFIQUES ===
    // ============================

    /** Recherche par catégorie (insensible à la casse) */
    public List<Amenity> getByCategory(String category) {
        return amenityRepository.findByCategoryIgnoreCase(category);
    }

    /** Recherche par nom partiel (insensible à la casse) */
    public List<Amenity> searchByName(String name) {
        return amenityRepository.findByNameContainingIgnoreCase(name);
    }


}

File: archive/front1-service/src/app/core/services/amenity.service.ts
// src/app/core/services/amenity.service.ts

import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { environment } from '../../../environments/environment';
import { Amenity } from '../models/amenity.model';

/**
 * ============================
 * SERVICE AMENITY
 * Gère les équipements (WiFi, Piscine, etc.)
 * ============================
 */
@Injectable({
  providedIn: 'root'
})
export class AmenityService {
  private readonly baseUrl = environment.services.listing; // '/listings'

  constructor(private apiService: ApiService) {}

  /**
   * ============================
   * RÉCUPÉRER TOUTES LES AMENITIES
   * GET /listings/amenities
   * ============================
   */
  getAllAmenities(): Observable<Amenity[]> {
    return this.apiService.get<Amenity[]>(`${this.baseUrl}/amenities/all`);
  }

  /**
   * ============================
   * RÉCUPÉRER PAR CATÉGORIE
   * GET /listings/amenities/category/{category}
   * ============================
   */
  getByCategory(category: string): Observable<Amenity[]> {
    return this.apiService.get<Amenity[]>(`${this.baseUrl}/amenities/category/${category}`);
  }

  /**
   * ============================
   * RECHERCHER PAR NOM
   * GET /listings/amenities/search?name={name}
   * ============================
   */
  searchByName(name: string): Observable<Amenity[]> {
    return this.apiService.get<Amenity[]>(`${this.baseUrl}/amenities/search`, { name });
  }
}
